src/components/IntervalField.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useMemo, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { useField } from 'formik'
+import { v4 } from 'uuid'
+import { Flex, Typography, SubdomainInput } from '@etvas/etvaskit'
+import { SubLabel } from './TagInput/SubLabel'
+
+export const IntervalField = ({
+  id,
+  name,
+  label,
+  placeholder,
+  error,
+  noPreserveSpace,
+  disabled,
+  separator,
+  suffix,
+  suffixSpace
+}) => {
+  const [{ value }, , { setValue }] = useField(name)
+
+  const [idLeft, idRight] = useMemo(() => {
+    const idOrNameId = id || `${name}-${v4()}`
+    return [`${idOrNameId}-left`, `${idOrNameId}-right`]
+  }, [id, name])
+
+  useEffect(() => {
+    if (!value.toString().includes('-')) {
+      setValue(`${value}-`)
+    }
+  }, [value, setValue])
+
+  const [placeholderLeft, placeholderRight] = useMemo(() => {
+    const split = placeholder?.split('-')
+    return [split?.shift(), split?.pop()]
+  }, [placeholder])
+
+  const [leftValue, rightValue] = useMemo(() => {
+    const split = value?.toString().split('-')
+    return [split?.shift(), split?.pop()]
+  }, [value])
```

Comment:
these 2 avoid memo if not expensive computation

Reply:
0cf6eecd84f58547138ab99772dfa7a580b8285b